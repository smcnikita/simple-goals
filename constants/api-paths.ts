export const API_PATHS = {
  REGISTER: '/register',
  GET_USER: '/user',
  GOALS: {
    GET: '/goals',
    CREATE_GOAL: '/goals',
    DELETE_GOAL: '/goals',
    UPDATE_GOAL: '/goals',
    UPDATE_SHOW_STATISTIC: '/goals/update-show-statistic',
    UPDATE_CAN_EDIT_PAST: '/goals/update-can-edit-past',
    MARK_ALL_AS_INCOMPLETE: '/goals/mark-all-incomplete',
    UPDATE_DESCRIPTION_SETTINGS: '/goals/update-description-settings',
  },
  CREATE_NEXT_YEAR: '/user/years/create-next',
  UPDATE_USER_NAME: '/user/update-name',
  UPDATE_USER_PASSWORD: '/user/update-password',
  DELETE_USER: '/user',
  ENCRYPT_GOALS: '/user/encrypt',
  DECRYPT_GOALS: '/user/decrypt',
  SECTIONS: {
    GET: '/sections',
    CREATE_SECTION: '/sections',
    DELETE_SECTION: '/sections',
    UPDATE_SECTION: '/sections',
  },
};
