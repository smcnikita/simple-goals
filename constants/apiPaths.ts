export const API_PATHS = {
  auth: {
    signIn: '/auth/login',
    signUp: '/auth/signup',
    logout: '/auth/logout',
  },

  goals: {
    update: '/goals/change',
    remove: '/goals',
  },
};
