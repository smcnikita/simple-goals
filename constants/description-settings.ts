export const DESCRIPTION_SETTINGS_KEYS = {
  display_none: 'display_none',
  display_3_lines: 'display_3_lines',
  display_full: 'display_full',
} as const;
