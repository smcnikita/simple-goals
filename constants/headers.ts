export const USER_ID = 'x-user-id';
