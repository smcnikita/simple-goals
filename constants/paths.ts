export const PATHS = {
  home: '/',

  auth: {
    signIn: '/sign-in',
    signUp: '/sign-up',
  },
};
