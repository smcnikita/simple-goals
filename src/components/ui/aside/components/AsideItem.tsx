import clsx from 'clsx';
import Link from 'next/link';
import { type FC, type PropsWithChildren } from 'react';

import classes from '../styles/item.module.css';

type Props = {
  value: string;
  isActive: boolean;
  href: string;
  className?: string;
};

const AsideItem: FC<PropsWithChildren<Props>> = ({ value, href, className, isActive }) => {
  return (
    <li className={classes.item}>
      <Link
        href={href}
        className={clsx(classes.link, className, {
          [classes.active]: isActive,
        })}
      >
        {value}
      </Link>
    </li>
  );
};

export default AsideItem;
