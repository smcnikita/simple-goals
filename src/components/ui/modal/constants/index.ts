import type { Styles } from 'react-modal';

export const DEFAULT_STYLES: Styles = {
  content: {
    position: 'static',
    width: '100%',
    maxWidth: '600px',
    margin: '0 24px',
    borderRadius: '12px',
    boxShadow: '0 4px 10px rgba(0, 0, 0, 0.3)',
    color: 'var(--body-color)',
    background: 'var(--body-background)',
    border: '1px solid rgba(0, 0, 0, 0.1)',
    padding: '32px 24px',
  },
  overlay: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    zIndex: 100,
    background: 'rgba(0, 0, 0, 0.5)',
    backdropFilter: 'blur(15px)',
  },
};
