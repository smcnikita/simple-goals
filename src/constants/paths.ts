export const PATHS = {
  home: '/',

  auth: {
    signIn: '/sign-in',
    signUp: '/sign-up',

    oauth: {
      github: '/oauth/callback/github',
      yandex: '/oauth/callback/yandex',
    },
  },

  goals: {
    base: '/goals',
    slug: '/:slug',
  },
};
