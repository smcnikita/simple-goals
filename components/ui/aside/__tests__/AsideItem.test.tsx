import { usePathname } from 'next/navigation';
import { render, screen } from '@testing-library/react';

import { PATHS } from '@/constants/paths';

import AsideItem from '../components/AsideItem';

jest.mock('next/navigation', () => ({
  usePathname: jest.fn(),
}));

describe('AsideItem', () => {
  const year = 2023;
  const href = PATHS.goals.base + PATHS.goals.slug.replace(':slug', year.toString());

  it('should renders AsideItem unchanged', () => {
    const { container } = render(<AsideItem year={year} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the year correctly', () => {
    (usePathname as jest.Mock).mockReturnValue('/some-other-path');

    render(<AsideItem year={year} />);

    expect(screen.getByText(year.toString())).toBeInTheDocument();
  });

  it('should apply active class when pathname matches href', () => {
    (usePathname as jest.Mock).mockReturnValue(href);

    render(<AsideItem year={year} />);

    const link = screen.getByRole('link', { name: year.toString() });
    expect(link).toHaveClass('active');
  });

  it('should generate correct href', () => {
    (usePathname as jest.Mock).mockReturnValue('/some-other-path');

    render(<AsideItem year={year} />);

    const link = screen.getByRole('link', { name: year.toString() });
    expect(link).toHaveAttribute('href', href);
  });
});
