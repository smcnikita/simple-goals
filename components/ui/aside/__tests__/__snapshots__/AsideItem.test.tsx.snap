// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideItem should renders AsideItem unchanged 1`] = `
<div>
  <li
    class="item"
  >
    <a
      class="link"
      href="/goals/2023"
    >
      2023
    </a>
  </li>
</div>
`;
